src/init.cpp

```diff
@@ -174,9 +174,20 @@ void Interrupt(NodeContext& node)
     }
 }
 
+#if HAVE_SYSTEM
+static void ShutdownNotify(const ArgsManager& args)
+{
+    std::string cmd = args.GetArg("-shutdownnotify", "");
+    if (!cmd.empty()) {
```

Comment:
Suggest using a loop over potentially multiple options as in #22372

Reply:
Done 👍🏻